### PR TITLE
Update/test checkout

### PR DESCRIPTION
**Pull request by okhan-okbay-cko**
This is a huge PR but it was necessary for this to go all in at once and most of the changes are folder path updates. 

- Removes some unused files
- Removes an additional cascaded layer of `Checkout/Checkout` into `Checkout` only
- Runs tests under `Checkout/Source` as part of the Bitrise workflows
- Makes all the sample projects work and the most importantly, all the samples now mimic the merchant behaviour with no local drag and dropping or referencing of the SDKs (i.e. they fetch the code from remote sources as our merchants do)
- `Package.swift` now denotes the package structure and to be used as a base for developments from now on

### CODE REVIEW

**Comment by deepesh-vasthimal-cko** [comment]
The Bitrise YML file might need an update as the build is failing with below error
`Checkout/Samples/CocoapodsSample/CheckoutCocoapodsSample.xcworkspace is not exist`

File: Checkout/Checkout.podspec [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Pod::Spec.new do |s|
+  s.name         = "Checkout"
+  s.summary      = "Checkout Core Features that can only be used with the right level of PCI Compliance. Please contact us for the details."
```
Comment (deepesh-vasthimal-cko) [inline Checkout/Checkout.podspec:3]: @patrick-hoban-cko Can you review this text? if already done please ignore.

File: screenshots/Package.swift [hunk 2]
```diff
@@ -0,0 +1,7 @@
+// swift-tools-version:5.5
+
+import PackageDescription
```
Comment (deepesh-vasthimal-cko) [inline screenshots/Package.swift:3]: why do the screenshots folder have `Package.swift`?

File: scripts/Package.swift [hunk 3]
```diff
@@ -0,0 +1,7 @@
+// swift-tools-version:5.5
+
+import PackageDescription
```
Comment (deepesh-vasthimal-cko) [inline scripts/Package.swift:3]: why does the scripts folder have Package.swift?

File: screenshots/Package.swift (hunk 2 above)
Comment (okhan-okbay-cko) [reply screenshots/Package.swift:3]: As mentioned in the PR description, it's to slim down and tidy up the distribution package. So, whenever an empty `Package.swift` file gets added to a folder like this, it excludes that folder from the SPM distribution. 

Merchants don't need to download our sample apps or screenshots or scripts to use our SDK. This is the only working way currently.

File: scripts/Package.swift (hunk 3 above)
Comment (okhan-okbay-cko) [reply scripts/Package.swift:3]: same as above

File: screenshots/Package.swift (hunk 2 above)
Comment (okhan-okbay-cko) [reply screenshots/Package.swift]: I removed these to avoid future confusion from our merchants' side

File: Checkout/Checkout.podspec (hunk 1 above)
Comment (patrick-hoban-cko) [reply Checkout/Checkout.podspec:3]: Checkout Core functionality can only be used with the right level of PCI Compliance. Please contact us for more details.

**Review by deepesh-vasthimal-cko — APPROVED** [review]

File: Checkout/Checkout.podspec [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  s.name         = "Checkout"
+  s.summary      = "Checkout Core functionality can only be used with the right level of PCI Compliance. Please contact us for more details."
+  s.version      = "4.1.0"
+  s.homepage     = "https://github.com/checkout/frames-ios"
+  s.swift_version = "5.0"
+  s.author       = { "Checkout.com Integration" => "integration@checkout.com" }
+  s.platform     = :ios, "12.0"
+  s.source       = { :path => "./Source" }
+  s.license      = "MIT"
+
```
[the remark this answers is not in the document]
Comment (okhan-okbay-cko) [reply Checkout/Checkout.podspec:11]: it's not deprecated but we'll continue releasing new versions of it. We can revisit to update this file when we need a new version. Probably we'd get some error/warnings in Terminal and just quickly update it

**Pull request merged**